test/operance_data_row_test.dart

```diff
@@ -0,0 +1,180 @@
+                  expansionBuilder: (item) => Text('Expanded $item'),
+                  onChecked: (_, {isSelected}) => checkedCalled = true,
+                  onRowPressed: (_) {},
+                  decoration: const OperanceDataDecoration(),
+                  isHovered: true,
+                  isSelected: true,
+                  isExpanded: true,
+                  showExpansionIcon: true,
+                  showCheckbox: true,
+                ),
+              ),
+            ),
+          );
+
+          expect(find.text('Test Row'), findsOneWidget);
+          expect(find.byType(Checkbox), findsOneWidget);
+          expect(find.byIcon(Icons.keyboard_arrow_down), findsOneWidget);
+          expect(find.text('Expanded Test Row'), findsOneWidget);
+
+          await tester.tap(find.byIcon(Icons.keyboard_arrow_down));
+          await tester.pumpAndSettle();
+
+          expect(expandedCalled, isTrue);
+
+          await tester.tap(find.byType(Checkbox));
+          await tester.pumpAndSettle();
+
+          expect(checkedCalled, isTrue);
+        },
+      );
+    });
+
+    group('When optional parameters are provided and row is not expanded', () {
+      testWidgets(
+        'Then it should create an instance with the provided values',
+        (tester) async {
+          var expandedCalled = false;
+          var checkedCalled = false;
+
+          await tester.pumpWidget(
+            MaterialApp(
+              home: Scaffold(
+                body: OperanceDataRow<String>(
+                  columnOrder: [0],
+                  columns: columns,
+                  row: 'Test Row',
+                  index: 0,
+                  tableWidth: 500.0,
+                  onEnter: (_) {},
+                  onExit: (_) {},
+                  onExpanded: (_) => expandedCalled = true,
+                  expansionBuilder: (item) => Text('Expanded $item'),
+                  onChecked: (_, {isSelected}) => checkedCalled = true,
+                  onRowPressed: (_) {},
+                  decoration: const OperanceDataDecoration(),
+                  isHovered: true,
+                  isSelected: true,
+                  isExpanded: false,
+                  showExpansionIcon: true,
+                  showCheckbox: true,
+                ),
+              ),
+            ),
+          );
+
+          expect(find.text('Test Row'), findsOneWidget);
+          expect(find.byType(Checkbox), findsOneWidget);
+          expect(find.byIcon(Icons.keyboard_arrow_left), findsOneWidget);
+
+          await tester.tap(find.byIcon(Icons.keyboard_arrow_left));
+          await tester.pumpAndSettle();
+
+          expect(expandedCalled, isTrue);
+
+          await tester.tap(find.byType(Checkbox));
+          await tester.pumpAndSettle();
+
+          expect(checkedCalled, isTrue);
+        },
+      );
+    });
+
+    group(
+        'When showExpansionIcon is true but onExpanded or expansionBuilder is '
+        'null', () {
```

Comment:
Could add comments here to explain why we are using test instead of test widgets
```suggestion
        'null', () {
          // Using `test` instead of `testWidgets` because we're only checking
          // for an assertion error, which doesn't require widget testing
```

Reply:
I'm not a fan of comments tbh, should only comment when code is complex, but  i think the then message is self descriptive as to why it's a regular test and not a widget one, if it's going to assert then i can't test the widget right